server/lib/notifications/notification-model.ts

```diff
@@ -0,0 +1,110 @@
+import sql from 'sql-template-strings'
+import { Notification, NotificationData } from '../../../common/notifications'
+import db from '../db/index'
+import { Dbify } from '../db/types'
+
+type DbNotification = Dbify<{
+  id: string
+  userId: number
+  read: boolean
+  visible: boolean
+  createdAt: Date
+  data: NotificationData
+}>
+
+function fromDbNotification(result: Readonly<DbNotification>): Notification {
+  return {
+    id: result.id,
+    type: result.data.type,
+    read: result.read,
+    createdAt: Number(result.created_at),
```

Comment:
We have definitely talked about this before, you do *not* want to do this conversion at this level. Keep this a Date until the very level at which it is transmitted to the client. You will likely need a type separate from the type returned/used by these methods (and not shared with the clients as "this is the type we transmit to you") to do so, and that is fine and normal